### ISSUE TITLE
mask and mean

### ISSUE THREAD

**Issue by Ryoo72** [LeeDoYup/FixMatch-pytorch#18]
안녕하세요! 작성해주신 코드 잘 보았습니다 !

아래의 링크에 걸린 부분에서, mask 를 곱해주신 뒤 mean을 이용해 평균을 내어주시는데, 이렇게 될 경우 마스크에 의해 값이 0 이 된 경우에도 평균을 낼 때 숫자가 count 되어 값이 좀 더 작아지는 현상이 있지 않나 궁금하여 질문을 드리게 되었습니다.
https://github.com/LeeDoYup/FixMatch-pytorch/blob/f167d6479b4ae109669b5bfa512ad3a23e572146/models/fixmatch/fixmatch_utils.py#L36

감사합니다 !

**Comment by LeeDoYup** [comment]
안녕하세요. 코드 이용해주셔서 감사합니다.
말씀하신 것이 맞고, 엄밀하게 구현하려면 0을 제외하고 count 하는 것이 맞습니다.

다만 이 코드를 남겨둔 이유는,
mask를 이용해서 샘플들을 제외하는 경우에는, 학습에서 사용하는 실제 배치사이즈가 줄어들게 됩니다.
이 때, 배치사이즈가 줄어드는 것에 비례하게 learning rate을 조절해주는 것이 자연스럽다고 생각해서 코드를 남겨놓았습니다.

감사합니다
